src/macaron/slsa_analyzer/checks/provenance_commit_check.py

```diff
@@ -0,0 +1,145 @@
+from macaron.slsa_analyzer.analyze_context import AnalyzeContext
+from macaron.slsa_analyzer.checks.base_check import BaseCheck
+from macaron.slsa_analyzer.checks.check_result import CheckResultData, CheckResultType, Confidence, JustificationType
+from macaron.slsa_analyzer.registry import registry
+from macaron.slsa_analyzer.slsa_req import ReqName
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class ProvenanceDerivedCommitFacts(CheckFacts):
+    """The ORM mapping for justifications in the commit from provenance check."""
+
+    __tablename__ = "_provenance_derived_commit_check"
+
+    #: The primary key.
+    id: Mapped[int] = mapped_column(ForeignKey("_check_facts.id"), primary_key=True)  # noqa: A003
+
+    #: The reason this test failed to pass, when applicable.
+    error: Mapped[str] = mapped_column(String, nullable=True, info={"justification": JustificationType.TEXT})
+
+    #: The state of the commit.
+    commit_digest: Mapped[str] = mapped_column(String, nullable=True, info={"justification": JustificationType.TEXT})
+
+    __mapper_args__ = {
+        "polymorphic_identity": __tablename__,
+    }
+
+
+class ProvenanceDerivedCommitCheck(BaseCheck):
+    """This check tries to extract the repo from the provenance and compare it to what is in the context."""
+
+    def __init__(self) -> None:
+        """Initialize instance."""
+        check_id = "mcn_provenance_derived_commit_1"
+        description = "Check whether the commit came from provenance."
+        depends_on: list[tuple[str, CheckResultType]] = []
+        eval_reqs = [ReqName.SECURITY]
+        super().__init__(
+            check_id=check_id,
+            description=description,
+            depends_on=depends_on,
+            eval_reqs=eval_reqs,
+            result_on_skip=CheckResultType.FAILED,
+        )
+
+    def run_check(self, ctx: AnalyzeContext) -> CheckResultData:
+        """Implement the check in this method.
+
+        Parameters
+        ----------
+        ctx : AnalyzeContext
+            The object containing processed data for the target repo.
+
+        Returns
+        -------
+        CheckResultData
+            The result of the check.
+        """
+        if ctx.dynamic_data["is_inferred_prov"]:
+            # Do not allow inferred provenance.
+            return CheckResultData(
+                result_tables=[
+                    ProvenanceDerivedCommitFacts(error="Provenance is inferred only.", confidence=Confidence.HIGH)
+                ],
+                result_type=CheckResultType.FAILED,
+            )
+
+        provenance = ctx.dynamic_data["provenance"]
+        if not provenance:
+            return CheckResultData(
+                result_tables=[ProvenanceDerivedCommitFacts(error="No provenance found.", confidence=Confidence.HIGH)],
+                result_type=CheckResultType.FAILED,
+            )
+
+        if not ctx.component or not ctx.component.repository:
```

Comment:
It seems to me that the type of `component`, `repository` and `commit_sha` are not nullable. I wonder if this is True as I have seen it being used as if it was non nullable here - https://github.com/oracle/macaron/blob/a57d282e3e0dec541464a45f0f8d1a9307512d91/src/macaron/slsa_analyzer/checks/build_service_check.py#L132
Therefore, I wonder if we need to check for None value here and down below for `commit_sha`.